### PR TITLE
dwarf: optimize dwarf parsing for speed

### PR DESCRIPTION
**Pull request by jacobly0**
This code is run when printing a stack trace in a debug executable, so it has to be fast even without compiler optimizations.

Adding a `@panic` to the top of `main` and running an x86_64 backend compiled compiler goes from `1m32.773s` to `0m3.232s`.

### CODE REVIEW

**Review by kubkon — APPROVED** [review]

**Comment by andrewrk** [comment]
Nice!

I want to note that the code used to operate on a file handle which is why it was originally written using seeking APIs and Readers. However now that it's based on memory mapping debug info, it's wasteful to use a streaming API  when it could just be doing, you know, slicing and indexing of random access memory buffers. Not that changing that needs to block this PR of course.

**Review by kubkon — APPROVED** [review]
Looks good! I wonder if custom zig extension will need special handling in the linkers, but we can cross that bridge once we get there.

**Comment by andrewrk** [comment]
CI has been unstable on master branch after merging this

**Comment by jacobly0** [comment]
> CI has been unstable on master branch after merging this

Looks like I got lucky and passed on zanic, but the c backend crashes compiling the code introduced by this PR on ziggy in particular.

**Pull request merged**
